pkg/cmd/roachtest/tests/restore.go

```diff
@@ -683,170 +656,48 @@ func (bs backupSpecs) CompatibleClouds() registry.CloudSet {
 	return r
 }
 
-func (bs backupSpecs) storagePrefix() string {
-	if bs.cloud == spec.AWS {
-		return "s3"
-	}
-	return "gs"
-}
-
-func (bs backupSpecs) backupCollection() string {
-	// N.B. AWS buckets are _regional_ whereas GCS buckets are _multi-regional_. Thus, in order to avoid egress (cost),
-	// we use us-east-2 for AWS, which is the default region for all roachprod clusters. (See roachprod/vm/aws/aws.go)
-	if bs.customFixtureDir != "" {
-		return bs.customFixtureDir
-	}
-	properties := ""
-	if bs.nonRevisionHistory {
-		properties = "/rev-history=false"
+func (bs *backupSpecs) CollectionURI(ctx context.Context, t test.Test) string {
+	if bs.collectionURI != "" {
+		return bs.collectionURI
 	}
-	switch bs.storagePrefix() {
-	case "s3":
-		return fmt.Sprintf(`'s3://cockroach-fixtures-us-east-2/backups/%s/%s/inc-count=%d%s?AUTH=implicit'`,
-			bs.workload.fixtureDir(), bs.version, bs.numBackupsInChain, properties)
-	case "gs":
-		return fmt.Sprintf(`'gs://cockroach-fixtures-us-east1/backups/%s/%s/inc-count=%d%s?AUTH=implicit'`,
-			bs.workload.fixtureDir(), bs.version, bs.numBackupsInChain, properties)
-	default:
-		panic(fmt.Sprintf("unknown storage prefix: %s", bs.storagePrefix()))
+	registry := GetFixtureRegistry(ctx, t, bs.cloud)
+	fixtureMeta, err := registry.GetLatest(ctx, bs.workload.Fixture().Kind())
+	if err != nil {
+		t.Skipf("fixture %s is not available: %s", bs.workload.Fixture().Kind(), err)
```

Comment:
My thinking here is that for non-`master` fixtures, after a new release, the initial fixtures may not have been created yet and restore tests will fail until they are created. However, this does mean that potential false negatives may occur where there may be an actual issue where the fixture is not accessible.